File: doc-examples/jdbc-example-java/src/main/java/example/Passenger.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.Index;
import io.micronaut.data.annotation.MappedEntity;

// tag::upsert-entity[]
@MappedEntity
@Index(columns = "email", unique = true)
public class Passenger {

    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    private Long id;

    private final String email;

    private String firstName;

    private String lastName;

    public Passenger(String email, String firstName, String lastName) {
        this.email = email;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
}
// end::upsert-entity[]

File: doc-examples/jdbc-example-python/src/main/python/example/Passenger.py
# tag::upsert-entity[]
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, Index, MappedEntity


@MappedEntity
@Index(columns=["email"], unique=True)
@dataclass
class Passenger:
    email: str
    firstName: str | None = None
    lastName: str | None = None
    id: Annotated[int | None, Id, GeneratedValue("IDENTITY")] = None
# end::upsert-entity[]
